crates/ra_hir/src/db.rs

```diff
@@ -31,6 +32,16 @@ pub trait HirDatabase: SyntaxDatabase
         use fn query_definitions::fn_syntax;
     }
 
+    fn struct_data(def_id: DefId) -> Cancelable<Arc<StructData>> {
+        type StructDataQuery;
+        use fn query_definitions::struct_data;
+    }
+
+    fn enum_data(def_id: DefId) -> Cancelable<Arc<EnumData>> {
```

Comment:
Note that we have a newtyped `FnId` for functions. I think we should either use type-safe def-ids everywhere, or a single DefId everywhere.

Reply:
Makes sense, yeah. I think I'll remove `FnId`, and will make `Module` a newtype of `DefId` as well.